components/forward_post_modal/forward_post_channel_select.tsx

```diff
@@ -0,0 +1,397 @@
+            description = `@${user.username}${deactivated}`;
+        } else {
+            name = user.username;
+            if (user.id === currentUserId) {
+                name += ` ${formatMessage({id: 'suggestion.user.isCurrent', defaultMessage: '(you)'})}`;
+            }
+            description = deactivated;
+        }
+    } else if (details.type === Constants.GM_CHANNEL) {
+        // remove the slug from the option
+        name = details.display_name;
+        description = '';
+    }
+
+    const sharedIcon = details.shared ? (
+        <SharedChannelIndicator
+            className='shared-channel-icon'
+            channelType={details.type}
+        />
+    ) : null;
+
+    const teamName = details.team_id && team ? (
+        <span className='option__team-name'>{team.display_name}</span>
+    ) : null;
+
+    return (
+        <div
+            id={`post-forward_channel-select_${details.name}`}
+            className='option'
+            data-testid={details.name}
+            aria-label={name}
+        >
+            {icon}
+            <span className='option__content'>
+                <span className='option__content--text'>{name}</span>
+                {(isPartOfOnlyOneTeam || details.type === Constants.DM_CHANNEL) && description && (
+                    <span className='option__content--description'>{description}</span>
+                )}
+                {customStatus}
+                {sharedIcon}
+                {tag}
+            </span>
+            {!isPartOfOnlyOneTeam && teamName}
+        </div>
+    );
+};
+
+const DropdownIndicator = (props: IndicatorProps<ChannelOption>) => {
+    return (
+        <components.DropdownIndicator {...props}>
+            <ChevronDownIcon
+                size={16}
+                color={'rgba(var(--center-channel-color-rgb), 0.64)'}
+            />
+        </components.DropdownIndicator>
+    );
+};
+
+type Props<O> = {
+    onSelect: (channel: ValueType<O>) => void;
+    value?: O;
+}
+
+function ForwardPostChannelSelect({onSelect, value}: Props<ChannelOption>) {
+    const {formatMessage} = useIntl();
+    const {current: provider} = useRef<SwitchChannelProvider>(new SwitchChannelProvider());
+
+    const getDefaultResults = () => {
+        let options: GroupedOption[] = [];
+
+        const handleDefaultResults = (res: ProviderResults) => {
+            options = [
+                {
+                    label: 'Recent',
+                    options: res.items.filter((item) => item.channel.type !== 'threads').map((item) => {
+                        const {channel} = item;
+                        return {
+                            label: channel.display_name || channel.name,
+                            value: channel.id,
+                            details: channel,
+                        };
+                    }),
+                },
+            ];
+        };
+
+        provider.fetchAndFormatRecentlyViewedChannels(handleDefaultResults);
```

Comment:
The logic between the two works somewhat the same, but not exactly the same, and that's when you end up with a lot of branching logic that makes the code harder to follow and reason about. It also means that a change to one feature can accidentally affect another one, like if we make the channel switcher sorting configurable, we probably don't want that to apply here as well.

As for the abusing logic part, the suggestion components are more complicated because they need to support things like matching only part of the entered text (as opposed to all the text here and in the channel switcher), and they also have logic for trying to replace partial autocomplete terms when an entry is selected. Neither of those are needed in the channel switcher, and there's been quite a few bugs in the channel switcher caused by those specifically. Also, they share CSS for the autocomplete list which causes even more issues.

And for this case, using a Provider without the rest of the suggestion components feels a bit like a less extreme version of using something like `React.Component` without the rest of `React` somehow. Yeah, you might be able to reuse some logic from it, but you're using it in a way it wasn't intended to work, and it might break down the line because of that

Reply:
ok, fair enough ... you convinced me to write my own logic and replace this with that. 👍🏼

Reply:
I guess this didn't end up happening then? At this point, I'd be okay letting this go in even though I'm still worried about this causing more problems down the line